### ISSUE TITLE
Keep track of local client

### ISSUE THREAD

**Issue by drumfreakk** [kraxarn/spotify-qt#100]
Spotify-qt should try to find existing local spotify clients, so that it doesnt start a new instance every time spotify-qt is started, as now every time you open and close spotify-qt a new instance of spotifyd is started

**Comment by kraxarn** [comment]
Did you try disabling "Always start" in "Settings > Spotify > General"?

**Comment by drumfreakk** [comment]
Ah, that checks whether there are any clients at all doesn't it, not just locally? I'd got that enabled because it didn't start automatically at all otherwise, but that's because I've got another client on my network that its probably recognizing. Would it still be something that would be worth implementing? I'd be happy to have a go at it.

**Comment by kraxarn** [comment]
Yeah, there is `spt::ClientHelper::isRunning` that checks if there's a local client running. Now that I actually look at it, the point of it is to never start the client if one is already running, so maybe something is actually wrong xd

**Comment by drumfreakk** [comment]
Might well be, I'll try and see if I can figure out whether its the code or just me

**Comment by kraxarn** [comment]
It currently uses `ps` to check for running processes. You could try running `ps aux | grep spotifyd` to see if it recognizes that it's running.

**Comment by shreyasminocha** [comment]
I have 'Always Start' and 'Start with app' enabled.

- Start with spotify-qt closed.
- Run `/usr/bin/spotifyd --bitrate 320 --username [redacted] --password '[redacted]' --no-daemon --device-name 'spotify-qt (spotifyd)' --backend pulseaudio`
- Open spotify-qt
- Run `ps aux | grep spotifyd`

      username  203850  0.2  0.2 365592 17300 pts/3    Sl+  10:26   0:00 /usr/bin/spotifyd --bitrate 320 --username [redacted] --password [redacted] --no-daemon --device-name spotify-qt (spotifyd) --backend pulseaudio
      username  204089  0.0  0.1 365332 12472 ?        Sl   10:27   0:00 /usr/bin/spotifyd --bitrate 320 --username [redacted] --password [redacted] --no-daemon --device-name spotify-qt (spotifyd) --backend pulseaudio
      username  204404  0.0  0.0   6756  2224 pts/5    S+   10:28   0:00 grep --color=auto spotifyd

Closing spotify-qt seems to usually kill the spotifyd process -qt started, but when it doesn't, re-opening -qt spawns a new spotifyd process.
